src/plugins/intel_npu/src/al/include/npuw_private_properties.hpp

```diff
@@ -0,0 +1,245 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/runtime/intel_npu/properties.hpp"
+
+namespace ov {
+namespace intel_npu {
+/**
+ * @brief
+ * Type: bool
+ * Set this option to true to utilize NPUW extension
+ * Possible values: true, false
+ * Default value: false
+ */
+static constexpr ov::Property<std::string> use_npuw{"NPU_USE_NPUW"};
+
+namespace npuw {
+/**
+ * @brief
+ * Type: std::string
+ * Device list to try in order.
+ * Possible values: Comma-separated list of devices. E.g., "NPU,GPU,CPU".
+ * Default value: "NPU,CPU"
+ */
+static constexpr ov::Property<std::string> devices{"NPUW_DEVICES"};
+
+/**
+ * @brief
+ * Type: std::string.
+ * Force the specific subgraph to specific device. The device must be present in the NPUW_DEVICES list.
+ * Possible values: Comma-separated "Subgraph index:OpenVINO device name" pairs,
+ * e.g. "0:CPU,1:NPU".
+ * Default value: empty.
+ */
+static constexpr ov::Property<std::string> submodel_device{"NPUW_SUBMODEL_DEVICE"};
+
+namespace partitioning {
+namespace online {
+/**
+ * @brief
+ * Type: std::string.
+ * Specify which partitioning pipeline to run.
+ * Possible values: "NONE", "INIT", "JUST", "REP".
+ * Default value: "REP".
+ */
+static constexpr ov::Property<std::string> pipeline{"NPUW_ONLINE_PIPELINE"};
```

Comment:
In the end, yes, but in Python we use strings to pass, and I believe it will remain like so.

In C++-ish GenAI those can be enums, but that's the only C++ code sensible to that which I am aware of (and we own it)